frontend/components/product-list/sections/FilterableProductsSection/RangeInput.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+   Box,
+   FormControl,
+   FormErrorMessage,
+   FormLabel,
+   HStack,
+   Input,
+   InputGroup,
+   InputLeftElement,
+   Text,
+} from '@chakra-ui/react';
+import { formatFacetName } from '@helpers/algolia-helpers';
+import { useDebounce } from '@ifixit/ui';
+import * as React from 'react';
+import { useRange, UseRangeProps } from 'react-instantsearch-hooks-web';
+import { useSearchCache } from './useSearchCache';
+
+export type RangeInputProps = UseRangeProps;
+
+const DEBOUNCE_INTERVAL_MILLIS = 300;
+
+export function RangeInput(props: RangeInputProps) {
+   const { refresh } = useSearchCache();
+   // This is a quick hack to get `useRange` to work. Without this, `useRange`
+   // would behave like price is not refinable. Eventually we should figure out
+   // how to make `useRange` work without this hack.
+   React.useEffect(() => {
+      refresh();
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+   }, []);
+
+   const range = useRange(props);
+   const appliedRangeMin = range.start[0];
+   const appliedRangeMax = range.start[1];
+
+   const refineMinValue = React.useCallback(
+      (value: number | undefined) => {
+         const minAllowedValue = range.range.min ? range.range.min + 1 : 0;
+         const maxAllowedValue = range.range.max
+            ? range.range.max - 1
+            : Infinity;
+         const clampedValue = clamp(value, minAllowedValue, maxAllowedValue);
+
+         const isValidRange =
+            clampedValue == null ||
+            appliedRangeMax == null ||
+            clampedValue < appliedRangeMax;
+
+         if (isValidRange) {
+            range.refine([
+               clampedValue,
+               Number.isFinite(appliedRangeMax) ? appliedRangeMax : undefined,
+            ]);
+         }
+      },
+      [appliedRangeMax, range]
+   );
+
+   const refineMaxValue = React.useCallback(
+      (value: number | undefined) => {
+         const maxAllowedValue = range.range.max
+            ? range.range.max - 1
```

Comment:
Pretty sure this makes for buggy behavior:

https://user-images.githubusercontent.com/26855/172952580-3d774a9e-db3a-4fdc-a76f-9da5ca8a151b.mp4

Reply:
The behavior is still a bit buggy and I feel like it's just trying to be too smart. I'd voter for just accepting the numbers they type in. Having numbers auto-change to something you didn't intend is surprising and a little annoying (and it accidentally hides the highest price item). Further searches may return different results and the auto-clamped values may then be irrelevant.

https://user-images.githubusercontent.com/26855/173158397-ccd20825-2b7c-48bb-8688-d6635fc43c8e.mp4

Reply:
I think let's merge this and fix it in a follow-up.